app/models/manageiq/providers/redhat/infra_manager/event_parser.rb

```diff
@@ -58,4 +58,57 @@ def self.ems_ref_from_object_in_event(data)
     return nil unless data.respond_to?(:[])
     ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(data[:href])
   end
+
+  def self.parse_new_target(full_data, message, ems)
+    cluster = full_data[:cluster]
+    cluster_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(cluster[:href])
+
+    cluster_name = nil
+    ems.with_provider_connection do |rhevm|
+      cluster_name = Ovirt::Cluster.find_by_href(rhevm, cluster_ref).try(:[], :name)
+    end
+
+    {
+      :ems_id         => ems.id,
+      :vm             => parse_new_vm(full_data[:vm], message),
+      :cluster        => parse_new_cluster(cluster_ref, cluster[:id], cluster_name),
+      :resource_pools => parse_new_rp(cluster[:id], cluster_name),
+      :folders        => parse_new_dc(full_data[:data_center], ems)
+    }
+  end
+
+  def self.parse_new_vm(vm, message)
+    ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(vm[:href])
+
+    ManageIQ::Providers::Redhat::InfraManager::RefreshParser.create_vm_hash(
+      ems_ref.include?('/templates/'), ems_ref, vm[:id], message.split(/\s/)[1])
+  end
+
+  def self.parse_new_cluster(cluster_ref, cluster_id, cluster_name)
+    {
+      :ems_ref     => cluster_ref,
```

Comment:
@Fryguy the `:ems_id` for the new target is given [here](https://github.com/ManageIQ/manageiq/pull/10934/files#diff-52fa558b219d05b5565f5fc8ea42e88cR72) so that should cover the rest of these (`vm`, `cluster`, etc...)